### ISSUE TITLE
Add support for pipenv

### ISSUE THREAD

**Issue by calebmeyer** [MasoniteFramework/masonite#95]
https://docs.pipenv.org/

Pipenv is essentially a better pip. It handles dependency resolution (of course), but also handles your virtual environments for you. It integrates with pyenv if you're using it to automatically get the correct python version for a project, and it auto-loads .env files (such as the ones masonite uses).

I'd love to see `craft install` run `pipenv install` under the hood.

**Comment by josephmancuso** [comment]
before the release of 2.0 it actually had full pipenv support. We could actually still add it back if you'd like. It would run pipenv if it was installed on the system. else it would default to pip.

After talking with the community we sort of decided to leave it until 2.1 because pipenv just sort of released a big version and started gettting a lot of attention so we weren't sure if we should do poetry or pipenv.

I think ADDING pipenv support would be awesome for now. Eventually we have to make Masonite either default to pipenv or Poetry. Whichever one wins the battle. Pipenv vs Poetry is really 50/50 and I don't want to force developers to have to choose one over the other but eventually we will have to make that decision.

I like poetry but pipenv plays MUCH better with Masonite and Craft.

If you want to work together to add pipenv support back to Craft then i'm open to talk about that for sure.

Pipenv allowed activating the virtual environment from a subprocess call so we were able to make the installation + virtual environment down to 2 commands:

```
$ craft install
$ craft serve
```

**Comment by josephmancuso** [comment]
I can try to find the commits that contained that logic. Hopefully i commited it

**Comment by josephmancuso** [comment]
Ill close this for now but this needs to be brought back up for the next release.

**Comment by gloc-mike** [comment]
Isn't **pipenv** the "officially recommended python packaging tool"?

Would be great to see it resurrected!

**Comment by josephmancuso** [comment]
yeah it's about that time to open the issue back up. Masonite will be coming out with 3 beta releases before the release in december. Beta 2 will likely have full pipenv support and we will see how it works and either keep or remove it for Beta 3
